### PR TITLE
Add option for forcing function creation (fixes #11165)

### LINKED ISSUE #11165: `queryInterface.createFunction` should default to `CREATE OR REPLACE` syntax.
## What are you doing?

```js
    queryInterface.createFunction(
            'is_timezone',
            [{ type: 'TEXT', name: 'tz', direction: 'IN' }],
            'BOOLEAN',
            'plpgsql',
            `
                PERFORM now() AT TIME ZONE tz;
                RETURN TRUE;
                EXCEPTION WHEN invalid_parameter_value THEN
                    RETURN FALSE;
            `,
            ['IMMUTABLE', 'LEAKPROOF'],
        );
```
**To Reproduce**
1. Call the above function twice, and you'll get an error

## What do you expect to happen?
Expecting following sql to run:
```sql
CREATE OR REPLACE FUNCTION is_timezone(IN tz TEXT) RETURNS BOOLEAN AS $func$ BEGIN 
  PERFORM now() AT TIME ZONE tz;
  RETURN TRUE;
  EXCEPTION WHEN invalid_parameter_value THEN
    RETURN FALSE;
END; $func$ language 'plpgsql'IMMUTABLE LEAKPROOF;
```

## What is actually happening?
```sql
CREATE FUNCTION is_timezone(IN tz TEXT) RETURNS BOOLEAN AS $func$ BEGIN 
  PERFORM now() AT TIME ZONE tz;
  RETURN TRUE;
  EXCEPTION WHEN invalid_parameter_value THEN
    RETURN FALSE;
END; $func$ language 'plpgsql'IMMUTABLE LEAKPROOF;
```
(Note the lack of `OR REPLACE` syntax)

## Environment
Dialect:
- [ ] mysql
- [x] postgres
- [ ] sqlite
- [ ] mssql
- [ ] any
Dialect **library** version: ^7.9.0
Database version: 9.6
Sequelize version: ^5.3.5
Node Version: 8.x
OS: Linux
Tested with latest release:
- [x] No
- [ ] Yes, specify that version:

### PR DESCRIPTION
**Pull request by mannol**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change
This adds an option to enable function replacement as described in #11165.

### CODE REVIEW

**Comment by mannol** [comment]
Some unrelated test failed ^. Can anyone retrigger?

**Comment by papb** [comment]
Nice work! Don't worry about the flaky test.

File: types/lib/query-interface.d.ts [hunk 1]
```diff
@@ -553,7 +557,8 @@ export class QueryInterface {
     returnType: string,
     language: string,
     body: string,
-    options?: QueryOptions
+    optionsArray?: string[],
+    options?: QueryOptionsWithForce
```
Comment (papb) [inline types/lib/query-interface.d.ts]: Looks like you also found (and fixed!) a bug in typings, right? Where it was previously `QueryOptions` you fixed it to `string[]`. Nice work.

Comment (mannol) [reply types/lib/query-interface.d.ts]: Yeah :+1:

**Pull request merged**